### DISCUSSION TITLE
When my app killed, My alarm app doesn't work anymore

### DISCUSSION THREAD

**Discussion by dugongcode** [gdelataillade/alarm#27] [General]
Hello. First of all, thank you for your dedication.I am currently developing an alarm app. We are developing alarm apps in many ways, and the biggest problem is that when the alarm app is completely turned off, sometimes the alarm goes off and sometimes it doesn't. I've tried many ways. Of course, I saw that the alarm doesn't ring when the alarm is killed even in the code guidance you made. But I wonder if you have any plans to improve this part in the future. I think it's too hard to force an alarm when the app is killed. Thanks once again.

**Comment by gdelataillade** [comment]
Hi @dugongcode 

Thanks for your interest in the package.

As explained in the README, when your app is killed, all background tasks are terminated so your alarm won't probably ring (it might depend on the Android version and manufacturer). Unfortunately, I think there is nothing I can do to fix this, because it is how the device operating system is designed.

The workaround I developed is to give you the possibility to show a notification at the moment the app is killed. The purpose of this notification is to warn the user that the app was killed and then all his alarms/timers are disabled. On the other hand, he can tap on this notification or manually reopen the app to reschedule automatically his alarms/timers.
